File: examples/javascript/calculate_taxes.js
/**
 * Calculate nanny taxes using the NannyKeeper API.
 *
 * Get a free API key: https://www.nannykeeper.com/developers/keys
 * Full docs: https://www.nannykeeper.com/developers
 */

const API_KEY = process.env.NANNYKEEPER_API_KEY || "nk_live_YOUR_KEY";
const BASE_URL = "https://www.nannykeeper.com/api/v1";

async function calculateTaxes(state, annualWages, payFrequency = "biweekly") {
  const response = await fetch(`${BASE_URL}/calculate`, {
    method: "POST",
    headers: {
      Authorization: `Bearer ${API_KEY}`,
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      state,
      annual_wages: annualWages,
      pay_frequency: payFrequency,
    }),
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(error.error?.message || `API error: ${response.status}`);
  }

  return (await response.json()).data;
}

async function checkThreshold(state, annualWages) {
  const response = await fetch(
    `${BASE_URL}/threshold?state=${state}&annual_wages=${annualWages}`,
    { headers: { Authorization: `Bearer ${API_KEY}` } }
  );

  if (!response.ok) {
    const error = await response.json();
    throw new Error(error.error?.message || `API error: ${response.status}`);
  }

  return (await response.json()).data;
}

// Example usage
async function main() {
  const data = await calculateTaxes("CA", 35000);

  console.log(`State: ${data.state_name}`);
  console.log(`Annual wages: $${data.annual_wages.toLocaleString()}`);
  console.log();
  console.log("Employer taxes (annual):");
  console.log(`  Social Security:    $${data.employer_taxes.social_security.toFixed(2)}`);
  console.log(`  Medicare:           $${data.employer_taxes.medicare.toFixed(2)}`);
  console.log(`  FUTA:               $${data.employer_taxes.futa.toFixed(2)}`);
  console.log(`  State unemployment: $${data.employer_taxes.state_unemployment.toFixed(2)}`);
  console.log(`  Total:              $${data.employer_taxes.total.toFixed(2)}`);
  console.log();
  console.log(`Per paycheck cost: $${data.per_paycheck.total_cost.toFixed(2)}`);

  // Check threshold
  console.log();
  const threshold = await checkThreshold("CA", 2500);
  console.log(`Threshold check ($2,500): ${threshold.federal.status}`);
  console.log(`  ${threshold.federal.explanation}`);
}

main().catch(console.error);

File: examples/python/calculate_taxes.py
"""
Calculate nanny taxes using the NannyKeeper API.

Get a free API key: https://www.nannykeeper.com/developers/keys
Full docs: https://www.nannykeeper.com/developers
"""

import os
import requests

API_KEY = os.environ.get("NANNYKEEPER_API_KEY", "nk_live_YOUR_KEY")
BASE_URL = "https://www.nannykeeper.com/api/v1"

def calculate_taxes(state: str, annual_wages: float, pay_frequency: str = "biweekly"):
    """Calculate household employer taxes for a given state and wage."""
    response = requests.post(
        f"{BASE_URL}/calculate",
        headers={"Authorization": f"Bearer {API_KEY}"},
        json={
            "state": state,
            "annual_wages": annual_wages,
            "pay_frequency": pay_frequency,
        },
    )
    response.raise_for_status()
    return response.json()["data"]

def check_threshold(state: str, annual_wages: float):
    """Check if wages cross the household employer threshold."""
    response = requests.get(
        f"{BASE_URL}/threshold",
        headers={"Authorization": f"Bearer {API_KEY}"},
        params={"state": state, "annual_wages": annual_wages},
    )
    response.raise_for_status()
    return response.json()["data"]

if __name__ == "__main__":
    # Calculate taxes for a nanny in California earning $35,000/year
    data = calculate_taxes("CA", 35000)

    print(f"State: {data['state_name']}")
    print(f"Annual wages: ${data['annual_wages']:,.0f}")
    print()
    print("Employer taxes (annual):")
    print(f"  Social Security:    ${data['employer_taxes']['social_security']:,.2f}")
    print(f"  Medicare:           ${data['employer_taxes']['medicare']:,.2f}")
    print(f"  FUTA:               ${data['employer_taxes']['futa']:,.2f}")
    print(f"  State unemployment: ${data['employer_taxes']['state_unemployment']:,.2f}")
    print(f"  Total:              ${data['employer_taxes']['total']:,.2f}")
    print()
    print(f"Per paycheck cost: ${data['per_paycheck']['total_cost']:,.2f}")

    # Check threshold
    print()
    threshold = check_threshold("CA", 2500)
    print(f"Threshold check ($2,500): {threshold['federal']['status']}")
    print(f"  {threshold['federal']['explanation']}")
